Userland/Applications/ThemeEditor/main.cpp

```diff
@@ -55,8 +47,18 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     auto window = GUI::Window::construct();
 
     auto main_widget = TRY(window->try_set_main_widget<ThemeEditor::MainWidget>());
-    if (path.has_value())
-        main_widget->load_from_file(TRY(Core::File::open(*path, Core::OpenMode::ReadOnly)));
+
+    if (path.has_value()) {
+        // Note: This is deferred to ensure that the window has already popped and thus proper window stealing can be performed.
+        app->event_loop().deferred_invoke(
+            [&window, &path, &main_widget]() {
+                auto response = FileSystemAccessClient::Client::the().try_request_file_read_only_approved(window, path.value());
+                if (response.is_error())
+                    return;
```

Comment:
Hmmm. Before if `ThemeEditor foo.ini` failed to load, ThemeEditor would close with an error, but now it silently ignores it. I'm not sure what the best option is but we should definitely let the user know that loading the file failed. Maybe just showing an error dialog would be good enough.

Reply:
A dialog seems reasonable!